app/models/ability.rb

```diff
@@ -146,57 +148,73 @@ def signed_in_with_roles(user)
     end
   end
 
-  def signed_in_with_organizer_role(user)
-    # ids of all the conferences for which the user has the 'organizer' role
-    conf_ids_for_organizer = Conference.with_role(:organizer, user).pluck(:id)
-
-    can :manage, Resource, conference_id: conf_ids_for_organizer
-    can [:new, :create], Conference if user.has_role?(:organizer, :any)
-    can :manage, Conference, id: conf_ids_for_organizer
-    can :manage, Splashpage, conference_id: conf_ids_for_organizer
-    can :manage, Contact, conference_id: conf_ids_for_organizer
-    can :manage, EmailSettings, conference_id: conf_ids_for_organizer
-    can :manage, Campaign, conference_id: conf_ids_for_organizer
-    can :manage, Target, conference_id: conf_ids_for_organizer
+  def signed_in_with_organization_admin_role(user)
+    org_ids_for_organization_admin = Organization.with_role(:organization_admin, user).pluck(:id)
+
+    can :manage, Organization, id: org_ids_for_organization_admin
+    can :new, Conference
+    can :manage, Conference, organization_id: org_ids_for_organization_admin
+    conf_ids_for_organization_admin = Conference.where(organization_id: org_ids_for_organization_admin).pluck(:id)
+    can [:index, :show], Role
+    can [:edit, :update], Role do |role|
+      role.resource_type == 'Organization' && (org_ids_for_organization_admin.include? role.resource_id)
+    end
+    signed_in_with_organizer_role(user, conf_ids_for_organization_admin)
+  end
+
+  def signed_in_with_organizer_role(user, conf_ids_for_organization_admin = [])
+    # ids of all the conferences for which the user has the 'organizer' role and
+    # conferences that belong to organizations for which user is 'organization_admin'
+    conf_ids = conf_ids_for_organization_admin.concat(Conference.with_role(:organizer, user).pluck(:id)).uniq
+    can :manage, Resource, conference_id: conf_ids
+    can :manage, Conference, id: conf_ids
```

Comment:
> If I run this on rails console:
> Ability.new(user_organizer).can? :new, Conference
> It returns true

That's right. Don't confuse the `:new` permission with checking permissions on `Conference.new` :)
The documentation says:

```ruby
Ability.new(user_organizer).can? :new, Conference # => true
Ability.new(user_organizer).can? :new, Conference.new # => false, this is what load_resource does
```

So the test has to check for the `:new` permission on `Conference.new`